### PR TITLE
Decouples starting gear from UtilityAI

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
Removes starting gear as a field in UtilityAI, allowing it to be specified independently is yaml. This was requested by Peptide citing issues with ghost roles/player controlling mobs with UtilityAI (centcomm official, etc).

**Screenshots**

**Changelog**

### CODE REVIEW

**Comment by metalgearsloth** [comment]
highly fucking based

File: Content.Server/Clothing/LoadoutSystem.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<LoadoutComponent, ComponentStartup>(OnStartup);
+        }
+
+        private void OnStartup(EntityUid uid, LoadoutComponent component, ComponentStartup args)
+        {
+            if (!_protoMan.TryIndex<StartingGearPrototype>(component.Prototype, out var proto))
+                return;
```
Comment (metalgearsloth) [inline Content.Server/Clothing/LoadoutSystem.cs:23]: This should Index or log that it failed.

File: Content.Server/Clothing/LoadoutSystem.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+using Content.Server.Clothing.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.Roles;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Clothing
+{
+    public sealed class LoadoutSystem : EntitySystem
```
Comment (metalgearsloth) [inline Content.Server/Clothing/LoadoutSystem.cs:8]: Comment

File: Content.Server/Clothing/Components/LoadoutComponent.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+using Content.Shared.Roles;
+
+namespace Content.Server.Clothing.Components
+{
+    [RegisterComponent]
+    public sealed class LoadoutComponent : Component
+    {
+        [DataField("prototype", required: true)]
```
Comment (metalgearsloth) [inline Content.Server/Clothing/Components/LoadoutComponent.cs:8]: typeserializer

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Ideally the job spawning code would also use this but future concern.

**Review by mirrorcult — APPROVED** [review]
besides sloth comments

**Comment by Peptide90** [comment]
For all intents and purposes fixes #6256 so ERT incoming! Good job.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**